src/napari_phasors/components_tab.py

```diff
@@ -0,0 +1,1001 @@
+            and self.component_line is not None
+        ):
+            layer = event.source
+            self.fractions_colormap = layer.colormap.colors
+            self.colormap_contrast_limits = layer.contrast_limits
+
+            self.draw_line_between_components()
+
+    def _on_contrast_limits_changed(self, event):
+        """Handle changes to the contrast limits of the fractions layer."""
+        if (
+            self.fractions_layer is not None
+            and self.component_line is not None
+        ):
+            layer = event.source
+            self.colormap_contrast_limits = layer.contrast_limits
+
+            self.draw_line_between_components()
+
+    def _make_components_draggable(self):
+        """Make component dots draggable."""
+        if self.parent_widget is None or self.drag_events_connected:
+            return
+
+        self.parent_widget.canvas_widget.canvas.mpl_connect(
+            'button_press_event', self._on_press
+        )
+        self.parent_widget.canvas_widget.canvas.mpl_connect(
+            'button_release_event', self._on_release
+        )
+        self.parent_widget.canvas_widget.canvas.mpl_connect(
+            'motion_notify_event', self._on_motion
+        )
+        self.drag_events_connected = True
+
+    def _on_press(self, event):
+        if event.inaxes is None:
+            return
+        for comp in self.components:
+            if comp.text is not None and comp.text.contains(event)[0]:
+                self.dragging_label_idx = comp.idx
+                return
+        for comp in self.components:
+            if comp.dot is not None and comp.dot.contains(event)[0]:
+                self.dragging_component_idx = comp.idx
+                return
+
+    def _on_motion(self, event):
+        if event.inaxes is None:
+            return
+        if self.dragging_label_idx is not None:
+            comp = self.components[self.dragging_label_idx]
+            if comp.text is not None and comp.dot is not None:
+                x, y = event.xdata, event.ydata
+                comp.text.set_position((x, y))
+                dx, dy = comp.dot.get_data()
+                comp.text_offset = (x - dx[0], y - dy[0])
+                self.parent_widget.canvas_widget.canvas.draw_idle()
+            return
+        if self.dragging_component_idx is None:
+            return
+        comp = self.components[self.dragging_component_idx]
+        if comp.dot is None:
+            return
+        x, y = event.xdata, event.ydata
+        comp.dot.set_data([x], [y])
+        comp.g_edit.setText(f"{x:.6f}")
+        comp.s_edit.setText(f"{y:.6f}")
+        if comp.text is not None:
+            ox, oy = comp.text_offset
+            comp.text.set_position((x + ox, y + oy))
+        self.draw_line_between_components()
+
+    def _on_release(self, event):
+        self.dragging_component_idx = None
+        self.dragging_label_idx = None
+
+    def _redraw(self, force=False):
```

Comment:
Are you needing to force redraw? This could go well in biaplotter in case it is a problem here.

Reply:
I mean, fine for this PR, but let me know and create an issue there if applicable :)